magda-web-client/src/Components/Dataset/Search/Facets/FacetTemporal.js

```diff
@@ -156,25 +156,19 @@ class FacetTemporal extends Component {
     }
 
     renderDatePicker() {
-        const temporalRangeStart = new Date(this.props.temporalRange[0]);
-        const temporalRangeEnd = new Date(this.props.temporalRange[1]);
-
-        const yearLower = temporalRangeStart.getUTCFullYear();
-        const monthLower = temporalRangeStart.getUTCMonth();
-
-        const yearUpper = temporalRangeEnd.getUTCFullYear();
-        const monthUpper = temporalRangeEnd.getUTCMonth();
-
+        if (
+            this.state.startYear > this.state.endYear ||
+            (this.state.startYear === this.state.endYear &&
+                this.state.startMonth > this.state.endMonth)
+        ) {
+            alert("End date is earlier than start date.");
```

Comment:
As discussed, it'd be good if this could be presented the same way as the `"Please enter a valid year."` message above :).

Reply:
The prompt style is similar to the "Please enter a valid year" now.
![prompt](https://user-images.githubusercontent.com/20970821/71940473-93442580-320a-11ea-9b27-b556b452f837.png)
Although, I noticed that the years/months don't get updated when typing is stopped :thinking: 
For example, if I finished typing 1987 (previous year was 2016), the end year is still 2016. But when I change 1987 to something else, the end year becomes 1987, so it's always late to register, if that makes sense :man_shrugging: 
But clicking the 'Apply' button does get the latest year. It just means that the prompt is late to come and late to go.

Reply:
I think the delay is ok (there's a slight debounce) but the error messages look different:

<img width="488" alt="image" src="https://user-images.githubusercontent.com/900555/71942790-7a3f7280-3212-11ea-92a1-047342602f54.png">

Reply:
Made it looks the same
![painful_css](https://user-images.githubusercontent.com/20970821/71952074-73742800-3231-11ea-8d32-54d5eef6663c.png)

Reply:
Tested just now and I'm able to go from an later year to an earlier year without that coming up:

<img width="478" alt="image" src="https://user-images.githubusercontent.com/900555/72028734-c7394c80-32d7-11ea-9e77-ded804cc7da0.png">

or https://2672-date-filter.dev.magda.io/search?dateFrom=3000-01-31T13%3A00%3A00.000Z&dateTo=2995-09-30T14%3A00%3A00.000Z&q=

Reply:
> Although, I noticed that the years/months don't get updated when typing is stopped thinking
For example, if I finished typing 1987 (previous year was 2016), the end year is still 2016. But when I change 1987 to something else, the end year becomes 1987, so it's always late to register, if that makes sense man_shrugging
But clicking the 'Apply' button does get the latest year. It just means that the prompt is late to come and late to go.

So what I meant here is not a time delay, but an actual delay in reading of the year (meaning, the props get the second-latest value of the years and months, not the very latest one). So when you stopped typing 1400, it didn't register 1400, it registered the year before 1400. If you change it again, 1400 will register and you should see the error. I am not sure what to do about this though :thinking:

Reply:
😂 oh right. Hmmm...